larpandoracontent/LArVertex/EnergyDepositionAsymmetryFeatureTool.cc

```diff
@@ -0,0 +1,193 @@
+
+    const CartesianVector localWeightedDirection(localWeightedDirectionSum.GetUnitVector());
+    const float evtProjectedVtxPos(vertexPosition2D.GetDotProduct(localWeightedDirection));
+
+    float minBeforeProjectedPos(std::numeric_limits<float>::max());
+    float maxBeforeProjectedPos(-std::numeric_limits<float>::max());
+
+    float minAfterProjectedPos(std::numeric_limits<float>::max());
+    float maxAfterProjectedPos(-std::numeric_limits<float>::max());
+
+    for (const VertexSelectionBaseAlgorithm::SlidingFitData &slidingFitData : slidingFitDataList)
+    {
+        const Cluster *const pCluster(slidingFitData.GetCluster());
+
+        CaloHitList caloHitList;
+        pCluster->GetOrderedCaloHitList().FillCaloHitList(caloHitList);
+
+        CaloHitVector caloHitVector(caloHitList.begin(), caloHitList.end());
+        std::sort(caloHitVector.begin(), caloHitVector.end(), LArClusterHelper::SortHitsByPosition);
+
+        for (const CaloHit *const pCaloHit : caloHitVector)
+        {
+            if (pCaloHit->GetPositionVector().GetDotProduct(localWeightedDirection) < evtProjectedVtxPos)
+            {
+                minBeforeProjectedPos = std::min(minBeforeProjectedPos, pCaloHit->GetPositionVector().GetDotProduct(localWeightedDirection));
+                maxBeforeProjectedPos = std::max(maxBeforeProjectedPos, pCaloHit->GetPositionVector().GetDotProduct(localWeightedDirection));
+
+                beforeVtxHitEnergy += pCaloHit->GetElectromagneticEnergy();
+                ++beforeVtxHitCount;
+            }
+
+            else
+            {
+                minAfterProjectedPos = std::min(minAfterProjectedPos, pCaloHit->GetPositionVector().GetDotProduct(localWeightedDirection));
+                maxAfterProjectedPos = std::max(maxAfterProjectedPos, pCaloHit->GetPositionVector().GetDotProduct(localWeightedDirection));
+
+                afterVtxHitEnergy += pCaloHit->GetElectromagneticEnergy();
+                ++afterVtxHitCount;
+            }
+        }
+    }
+
+    // Use energy metrics if possible, otherwise fall back on hit counting.
+    const unsigned int totHitCount(beforeVtxHitCount + afterVtxHitCount);
+
+    const float beforeVtxEnergyDeposition(
+        !std::fabs(maxBeforeProjectedPos - minBeforeProjectedPos) ? 0 : beforeVtxHitEnergy / std::fabs(maxBeforeProjectedPos - minBeforeProjectedPos));
+
+    const float afterVtxEnergyDeposition(
+        !std::fabs(maxAfterProjectedPos - minAfterProjectedPos) ? 0 : afterVtxHitEnergy / std::fabs(maxAfterProjectedPos - minAfterProjectedPos));
+
+    const float totalEnergyDeposition(beforeVtxEnergyDeposition + afterVtxEnergyDeposition);
+
+    if (useEnergyMetrics && totalEnergyDeposition)
+        return std::fabs((afterVtxEnergyDeposition - beforeVtxEnergyDeposition)) / totalEnergyDeposition;
+
+    if (0 == totHitCount)
+        throw StatusCodeException(STATUS_CODE_FAILURE);
+
+    const float beforeVtxHitDeposition(
```

Comment:
Hey Henry, this is certainly just me being extra picky but I see a minor inconsistency in the naming convention here... "After vertex hit energy" becomes "After vertex energy deposition" when dividing by the axis length, but "After vertex hit count" becomes "after vertex hit deposition". I also wonder if "deposition" is the most descriptive name that could be used here?

Reply:
Thanks for taking a look Maria! Yeah you're right it probably could be a little more consistent here. I'll try and wrap this up with the comments on linking the tool to the `GlobalAsymmetryFeatureTool`.